src/zenml/cli/stack.py

```diff
@@ -685,12 +685,21 @@ def rename_stack(
 
 
 @stack.command("list")
-def list_stacks() -> None:
-    """List all available stacks."""
+@click.option("--just_mine", "-m", is_flag=True, required=False)
+def list_stacks(just_mine: str = False) -> None:
+    """List all available stacks.
+
+    Args:
+        just_mine: To list only the stacks that the current user has created.
+    """
     cli_utils.print_active_config()
 
     client = Client()
     stacks = client.stacks
```

Comment:
If you look at the implementation of `client.stacks`, that method is first querying all your stacks, then all shared stacks, and then it combines them. Instead of then taking the shared stacks out again if `just_mine`, I'd suggest to not add the shared stacks in the first place. Perhaps this could be done by splitting the `client.stacks` logic into two separate methods `get_my_stacks()` and `get_shared_stacks()`, then you could just call `client.get_my_stacks()` directly.